packages/core/src/external/aws/lambda.ts

```diff
@@ -9,3 +11,86 @@ export function logResultToString(logResult: string | undefined): string | undef
   if (!logResult) return logResult;
   return base64ToString(logResult);
 }
+
+/**
+ * Checks whether the lambda invocation was successful.
+ */
+export function isLambdaError(
+  result: PromiseResult<AWS.Lambda.InvocationResponse, AWS.AWSError>
+): boolean {
+  return result.FunctionError !== undefined;
+}
+
+export type LambdaError = {
+  errorType: string;
+  errorMessage: string;
+  log: string | undefined;
+};
+
+/**
+ * Returns the error details if the lambda invocation failed.
+ */
+export function getLambdaError(
+  result: PromiseResult<AWS.Lambda.InvocationResponse, AWS.AWSError>
+): LambdaError | undefined {
+  if (!result.Payload) return undefined;
+  if (!isLambdaError(result)) return undefined;
+  const response = JSON.parse(result.Payload.toString());
+  return {
+    errorType: response.errorType,
+    errorMessage: response.errorMessage,
+    log: logResultToString(result.LogResult),
+  };
+}
+
+/**
+ * Returns a string representation of the payload if the lambda invocation was successful.
+ * If the lambda invocation failed, it throws a MetriportError (unless `failGracefully` is true,
+ * in which case it returns `undefined`).
+ *
+ * @param result The result of the lambda invocation
+ * @param lambdaName The name of the lambda that was invoked, used on error reporting (optional)
+ * @param failGracefuly If true, the function will return `undefined` instead of throwing an
+ *        error (optional, defaults to `false` - throw an error on failure)
+ * @param log A function to log errors (optional, defaults to `console.log`)
+ */
+export function getLambdaResultPayload(params: {
+  result: PromiseResult<AWS.Lambda.InvocationResponse, AWS.AWSError>;
+  lambdaName?: string;
+  failGracefuly?: boolean | false;
+  log?: typeof console.log;
+}): string;
+export function getLambdaResultPayload(params: {
+  result: PromiseResult<AWS.Lambda.InvocationResponse, AWS.AWSError>;
+  lambdaName?: string;
+  failGracefuly: true;
+  log?: typeof console.log;
+}): string | undefined;
```

Comment:
why are these 2 exports needed if there's the one below?

Reply:
Alloes for calls w/ diff params to have diff response types